File: 3812-smallest-palindromic-rearrangement-i/smallest-palindromic-rearrangement-i.java
class Solution {
    public String smallestPalindrome(String s) {
        int[] count = new int[26];
        for (char c : s.toCharArray()) {
            count[c - 'a']++;
        }
        StringBuilder leftHalf = new StringBuilder();
        String middleChar = "";
        
        for (int i = 0; i < 26; i++) {
            char c = (char) ('a' + i);
            int freq = count[i];
            
            for (int j = 0; j < freq / 2; j++) {
                leftHalf.append(c);
            }
            if (freq % 2 != 0) {
                middleChar = String.valueOf(c);
            }
        }
        String rightHalf = new StringBuilder(leftHalf).reverse().toString();
        return leftHalf.toString() + middleChar + rightHalf;
    }
}

File: 3812-smallest-palindromic-rearrangement-i/smallest-palindromic-rearrangement-i.py
class Solution:
    def smallestPalindrome(self, s: str) -> str:
        count = [0] * 26
        for char in s:
            count[ord(char) - ord('a')] += 1
            
        left_half = []
        middle_char = ""
        
        for i in range(26):
            c = chr(ord('a') + i)
            freq = count[i]
            
            left_half.append(c * (freq // 2))
            
            if freq % 2 != 0:
                middle_char = c
                
        left_str = "".join(left_half)
        return left_str + middle_char + left_str[::-1]